File: SumOfDiagonal.c
//Write a program to print the sum of the diagonal element of the M*N square matrix.

#include <stdio.h>

int main() {
    int n,i,j;
    printf("Enter size of square matrix (n x n): ");
    scanf("%d", &n);

    int matrix[n][n];
    printf("Enter matrix elements:\n");
    for (i = 0; i < n; i++)
        for (j = 0; j < n; j++)
            scanf("%d", &matrix[i][j]);

    int primarySum = 0, secondarySum = 0;

    for (i = 0; i < n; i++) {
        primarySum += matrix[i][i];               // Primary diagonal
        secondarySum += matrix[i][n - 1 - i];      // Secondary diagonal
    }

    printf("Sum of primary diagonal = %d\n", primarySum);
    printf("Sum of secondary diagonal = %d\n", secondarySum);

    return 0;
}

File: Sum_of_diagonal.py
#sum of diagonal

def diagonal_sums(matrix):
    n = len(matrix)
    primary = sum(matrix[i][i] for i in range(n))
    secondary = sum(matrix[i][n - 1 - i] for i in range(n))
    return primary, secondary

# Example square matrix
matrix = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
]

print("Matrix:")
for row in matrix:
    print(row)

primary, secondary = diagonal_sums(matrix)
print("Sum of primary diagonal =", primary)
print("Sum of secondary diagonal =", secondary)
